net: config: sntp: optionally set rtc

jeppenodgaard opened the pull request:
Set RTC when an SNTP response is received if enabled via Kconfig option.

Commit:
net: config: sntp: optionally set rtc

Set RTC when an SNTP response is received if enabled via Kconfig option.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

pdgendt commented on subsys/net/lib/config/init_clock_sntp.c:
```diff
@@ -46,7 +48,39 @@ static int sntp_init_helper(struct sntp_time *tm)
+static void sntp_set_rtc(const struct timespec *tspec)
+{
+#ifdef CONFIG_NET_CONFIG_CLOCK_SNTP_SET_RTC
+	const struct device *dev = DEVICE_DT_GET(DT_ALIAS(rtc));
```
```suggestion
	const struct device *dev = DEVICE_DT_GET(DT_CHOSEN(rtc));
```

jeppenodgaard commented:
Thanks for the review and inputs @pdgendt!
I added your suggestions :)

Also changed `gm_time` to `gm_time_r` because it is thread safe and there are no drawbacks AFAICT.

jeppenodgaard commented on subsys/net/lib/config/Kconfig:
```diff
@@ -215,6 +215,13 @@ config NET_CONFIG_CLOCK_SNTP_INIT
+config NET_CONFIG_CLOCK_SNTP_SET_RTC
+	bool "Set RTC on SNTP response"
+	depends on RTC
+	depends on $(dt_alias_enabled,rtc)
```
Should the chosen node be `zephyr,rtc = &my_rtc`...?

maass-hamburg reviewed (dismissed):
LGTM, maybe we could later spin this of into it's own responsible to sync with multiple time sources (adding gps or cellular as sources).
